### PR TITLE
Modify Find Command to work with students and lessons

### PR DESCRIPTION
**Pull request by pzl111**
##  Modify Find Command
Find command now works with students and lessons
* Modify find command to allow for both finding student and lesson using the same command
* Constructor takes in a string of trimmed arguments instead of a predicate
* During execution, depending on the model state, the correct predicate is created, and command executes either filtering student of lesson
* Change equals method in find command to check against the trimmed args, since predicate is not initialised until execution
* Messages.java: Create new output message when finding lessons

## Find command testing
* Add extra typical lessons for testing purposes
* Change find command tests to use trimmed args instead of predicates
* Add test cases to search for one, none, and multiple lessons

## Edit LessonCard
* Create new field in fxml for lesson name, and controller class to set the name
image

## Display lesson name in lesson details
* Create new lesson name field in lesson detail panel
image

## Bug fix with lesson names being reset to default name
* Lesson did not have a name, and cannot be used for display, also lesson name is saved as default name without getter
* Include name field, method to return name string for displaying, getter and setter for name

## Modify List Command
* Reset schedule list every time during list command
* Change list command to always remove filter past filter and give all persons and lessons when called

### CODE REVIEW

File: src/main/java/seedu/address/model/lessons/Lesson.java [hunk 1]
```diff
@@ -180,6 +189,19 @@ public TaskList getTaskList() {
         return taskList;
     }
 
+    public Name getName() {
+        return name;
```
Comment (yiwen101) [inline src/main/java/seedu/address/model/lessons/Lesson.java:193]: Thank you for fixing my leftover bug

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 2]
```diff
@@ -54,24 +55,64 @@ public void equals() {
...
+        model.setState(State.SCHEDULE);
+        expectedModel.updateFilteredScheduleList(new LessonContainsKeywordsPredicate(trimmedArgs));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_searchOneLesson() {
+        String expectedMessage = String.format(MESSAGE_LESSONS_LISTED_OVERVIEW, 1);
+        String trimmedArgs = "random";
+        FindCommand command = new FindCommand(trimmedArgs);
```
Comment (yiwen101) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:81]: Impressively robust

**Review by yiwen101 — APPROVED** [review]
LGTM
